jvector-base/src/main/java/io/github/jbellis/jvector/graph/GraphSearcher.java

```diff
@@ -322,15 +312,20 @@ public SearchResult resume(int additionalK, float threshold, float rerankFloor)
                 continue;
             }
 
-            // add its neighbors to the candidates queue
-            for (var it = view.getNeighborsIterator(topCandidateNode); it.hasNext(); ) {
-                int friendOrd = it.nextInt();
+            if (scoreFunction.supportsBulkSimilarity()) {
```

Comment:
Future work: let's see if we can update the ScoreFunction API to allow encapsulating this instead of making caller do a bunch of special cases